docs/hub/security-pickle.md

```diff
@@ -0,0 +1,226 @@
+# Pickle Scanning
+
+## What is a pickle ?
+
+From the [official docs](https://docs.python.org/3/library/pickle.html) :
+
+> The `[pickle](https://docs.python.org/3/library/pickle.html#module-pickle)` module implements binary protocols for serializing and de-serializing a Python object structure.
+> 
+
+What this means is that pickle is a serializing protocol, something you use to efficiently share data amongst parties.
+
+We call a pickle the binary file that was generated while pickling.
+
+At its core, the pickle is basically a stack of instructions or opcodes. As you probably have guessed, it’s not human readable. The opcodes are generated when pickling and read sequentially at unpickling. Based on the opcode, a given action is executed.
+
+Here’s a small example :
+
+```python
+import pickle
+import pickletools
+
+var = "data I want to share with a friend"
+
+# store the pickle data in a file named 'payload.pkl'
+with open('payload.pkl', 'wb') as f:
+    pickle.dump(var, f)
+
+# disassemble the pickle
+# and print the instructions to the command line
+with open('payload.pkl', 'rb') as f:
+    pickletools.dis(f)
+```
+
+When you run this, it will create a pickle file and print the following instructions in your terminal :
+
+```python
+    0: \x80 PROTO      4
+    2: \x95 FRAME      48
+   11: \x8c SHORT_BINUNICODE 'data I want to share with a friend'
+   57: \x94 MEMOIZE    (as 0)
+   58: .    STOP
+highest protocol among opcodes = 4
+```
+
+Don’t worry too much about the instructions for now, just know that the [pickletools](https://docs.python.org/3/library/pickletools.html) module is very useful for analyzing pickles. It allows you to read the instructions in the file ***without*** executing any code.
+
+Pickle is not simply a serialization protocol, it allows more flexibility by giving the ability to users to run python code at de-serialization time. Doesn’t sound good, does it ?
+
+## Why is it dangerous ?
```

Comment:
```suggestion
## Why is it dangerous?
```

Reply:
The French in me, fixing it for all punctuation !